label_studio/data_manager/managers.py

```diff
@@ -496,12 +510,60 @@ def __init__(self, expression, distinct=False, output_field=None, **extra):
 
 
 def annotate_completed_at(queryset):
+    if flag_set('fflag_feat_optic_161_project_settings_for_low_agreement_threshold_score_short'):
+        return annotated_completed_at_considering_agreement_threshold(queryset)
+
     from tasks.models import Annotation
 
     newest = Annotation.objects.filter(task=OuterRef('pk')).order_by('-id')[:1]
     return queryset.annotate(completed_at=Case(When(is_labeled=True, then=Subquery(newest.values('created_at')))))
 
 
+def annotated_completed_at_considering_agreement_threshold(queryset):
+    from tasks.models import Annotation
+
+    LseProject = load_func(settings.LSE_PROJECT)
+    get_tasks_agreement_queryset = load_func(settings.GET_TASKS_AGREEMENT_QUERYSET)
+
+    if not queryset.exists():
+        # No tasks to annotate. Quit early
+        return queryset
+
+    project_id = queryset[0].project_id
+
+    newest_annotation = Annotation.objects.filter(task=OuterRef('pk')).order_by('-id')[:1]
+    if not LseProject:
+        # Not LSE so there will not be agreement_threshold-based task completeness
+        return queryset.annotate(
+            completed_at=Case(When(is_labeled=True, then=Subquery(newest_annotation.values('created_at'))))
+        )
+
+    lse_project = LseProject.objects.filter(project_id=project_id).first()
+    agreement_threshold = lse_project.agreement_threshold
```

Comment:
is this possibly attempting to call against None?

maybe guard it just in case and drop the need to check lse_project first in the not below
```suggestion
    agreement_threshold = lse_project.agreement_threshold if lse_project else None
```